### PR TITLE
[core] Text renderer with columns

### PR DESCRIPTION
**Pull request by sgorbaty**
As a part of a plugin for Atom build script to run PMD I need access to columns, which default text renderer doesn't produce. This PR is to add that to a new AtomText renderer.

### CODE REVIEW

**Comment by jsotuyod** [comment]
Thanks for your time and dedication. Building support for more tools and IDEs is a top priority for us.

However, I have two concerns with this PR as it stands:
1. The name of the renderer. The fact that you are using it for an Atom plugin is circumstantial, not defining for the renderer itself. For an `AtomRenderer` class I would expect it to produce atom rss output.
1. I'm not fully convinced this needs an extra renderer. Atom uses Javascript as far as I know for extensions, parsing an xml report would be just as easy with JS using DOM / XPath / CSS Selectors. Even if not, maybe breaking backwards compatibility for text renderer on 5.6.0 is something we can live with, but thats my opinion, not necessarily shared by @adangel

**Comment by sgorbaty** [comment]
Hi @jsotuyod, my goal was to add a text renderer that outputs columns since the default 'text' renderer only prints the line #. If you could guide me through adding a text renderer with columns it would be greatly appreciated.

**Comment by jsotuyod** [comment]
@adangel what do you think? Can we break text reporter format? Or shall we insist on using another existing format such as xml?

**Comment by adangel** [comment]
@sgorbaty Thanks for the pull request. I agree with @jsotuyod , that we should not name it AtomRenderer. Instead, it is almost exactly like the TextRenderer, but with the columns in the output. I think, it makes sense, to enhance the existing TextRenderer for this. Then you also get processing errors and suppressed violations in the output.

So, now the question about compatibility - yes, we can break it with 5.6.0 (but definitely not with 5.5.3). Should we? I don't know. For me, it makes sense, to have the column by default in the output of the `TextRenderer` - it feels like it is just missing. If someone can't deal with it, we could either force them to adjust or we provide a backwards compatible switch. I would vote for providing a switch.

There are two ways for the switch:

1. Provide a second renderer ("TextRendererWithoutColumn"). That would be a subclass of TextRenderer. You can get an idea of how this works with the CSV renderers of CPD: [CSVRenderer](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/cpd/CSVRenderer.java) and [CSVWithLinecountPerFileRenderer](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/cpd/CSVWithLinecountPerFileRenderer.java).

2. The second way (which I would prefer): Since the renders for PMD support properties (they are all subclasses of [AbstractPropertySource](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/AbstractPropertySource.java), we should add a property to the TextRenderer, so that one can switch the column output off and on. I would have put it on by default for 5.6.0. If there are many complaints, we can decide for 5.6.1 if we switch it off by default, to restore the old behavior.
This property can be defined on the commandline with `-property showColumns=false`.

So, in summary:
1. Enhance the existing `TextRenderer` and add a `BooleanProperty` with the name "showColumns".
2. Make the default for the property "true".
3. Create a unit test like you already have. For the existing test, I guess you need to set the property in `getRenderer` before you return the renderer.

**Comment by jsotuyod** [comment]
@sgorbaty do you need any further help to move this forward?

**Comment by sgorbaty** [comment]
@jsotuyod It's not a priority, I'll probably get on this after holidays.
